astropy/cosmology/flrw.py

```diff
@@ -671,10 +631,10 @@ def nu_relative_density(self, z):
         # The massive and massless contribution must be handled separately
         # But check for common cases first
         if not self._massivenu:
-            if np.isscalar(z):
-                return prefac * self._Neff
-            else:
+            if getattr(z, "shape", ()):
```

Comment:
I realize this was here before, but why insist on returning something with the same shape? Broadcasting will take care of it anyway. If you do want it, maybe `np.full(np.shape(z), prefac * self._Neff)`

Reply:
I completely agree.
I'm working on a followup PR to address all the output shapes / types / etc.
For here and now, I'm trying not to change any outputs, only how the inputs are handled.